### PR TITLE
Add TOSA to TTIR conversions for sum and max reductions

### LINKED ISSUE #1430: Add TOSA to TTIR conversion for ReduceMaxOp
[tosa.reduce_max](https://mlir.llvm.org/docs/Dialects/TOSA/#tosareduce_max-mlirtosareducemaxop) lowers to ttir.max

### LINKED ISSUE #1479: Add TOSA to TTIR conversion for ReduceSumOp (sum reduction)
[tosa.reduce_sum](https://mlir.llvm.org/docs/Dialects/TOSA/#tosareduce_sum-mlirtosareducesumop) lowers to ttir.sum

### PR DESCRIPTION
**Pull request by sgligorijevicTT**
closes #1430 closes #1479

### CODE REVIEW

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 1]
```diff
@@ -80,6 +80,30 @@ class TosaToTTIRMultiplyOpConversionPattern
     return success();
   }
 };
+template <typename SrcOp, typename DestOp,
```
Comment (kmitrovicTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:83]: Blank line above.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 2]
```diff
@@ -80,6 +80,30 @@ class TosaToTTIRMultiplyOpConversionPattern
...
+          typename Adaptor = typename SrcOp::Adaptor>
+class TosaToTTIRReduceOpConversionPattern : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
```
Comment (kmitrovicTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:92]: Remove blank line.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 3]
```diff
@@ -80,6 +80,30 @@ class TosaToTTIRMultiplyOpConversionPattern
...
+        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+    rewriter.replaceOpWithNewOp<DestOp>(
+        srcOp, outputTensor.getType(), adaptor.getInput(), outputTensor,
+        true /*keepdim*/,
+        rewriter.getArrayAttr(SmallVector<Attribute>(1, adaptor.getAxisAttr())),
+        rewriter.getArrayAttr(
+            SmallVector<Attribute>(adaptor.getOperands().size() + 1,
+                                   rewriter.getAttr<OperandConstraintAttr>(
+                                       OperandConstraint::AnyDeviceTile))));
+    return success();
```
Comment (kmitrovicTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:104]: Feel free to use some blank lines between long statements like these to remove visual clutter.

File: test/ttmlir/Conversion/TosaToTTIR/reductions/sum.mlir [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// RUN: ttmlir-opt --convert-tosa-to-ttir %s | FileCheck %s
+module attributes {} {
+  func.func @test_sum(%arg0: tensor<13x21x3xf32>) -> tensor<13x1x3xf32> {
+    // CHECK: func.func {{.+}} [[IN_SIZE:tensor<[0-9]+x[0-9]+x[0-9]+xf32>]]{{.*}} ->
```
Comment (kmitrovicTT) [inline test/ttmlir/Conversion/TosaToTTIR/reductions/sum.mlir:4]: Hm, this wouldn't be too bad to have in many other tests which use hardcoded tensor shapes based on inputs. 

Please raise an issue to go through all tests in this repo and properly manage `CHECK`s like you did here.

**Review by kmitrovicTT — APPROVED** [review]
Overall good, minor comments below

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp (hunk 2 above)
Comment (sgligorijevicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: added

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp (hunk 1 above)
Comment (sgligorijevicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:84]: fixed

**Review by mrakitaTT — APPROVED** [review]

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 5]
```diff
@@ -81,6 +81,31 @@ class TosaToTTIRMultiplyOpConversionPattern
...
+          typename Adaptor = typename SrcOp::Adaptor>
+class TosaToTTIRReduceOpConversionPattern : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto outputType = mlir::cast<RankedTensorType>(srcOp.getResult().getType());
+    auto outputTensor = rewriter.create<tensor::EmptyOp>(
```
Comment (mrakitaTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:94]: Forgot to comment, let's try avoiding using auto whenever we can, except when type is very complicated. You can use `RankedTensorType` and `tensor::EmptyOp` for these two autos.

Comment (sgligorijevicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: fixed

**Pull request merged**